src/orb.yml.hbs

```diff
@@ -635,6 +635,9 @@ commands:
 
             # Prepare container definitions
             CONTAINER_DEFS=$(python $UPDATE_CONTAINER_DEFS_SCRIPT_FILE "$PREVIOUS_TASK_DEFINITION" "$CONTAINER_IMAGE_NAME_UPDATES" "$CONTAINER_ENV_VAR_UPDATES")
+            
+            # Escape single quotes from environment variables for BASH_ENV
+            CLEANED_CONTAINER_DEFS=$(sed -r "s/'+/'\\\''/g" \<<< $CONTAINER_DEFS)
```

Comment:
What do you think of `CLEANED_CONTAINER_DEFS=$(echo $CONTAINER_DEFS | sed -E "s/'+/'\\\''/g")` for greater compatibility with different `sed` distributions and POSIX standards?

This solution seems to work really well except for one case, which is, when `CONTAINER_DEFS` contains more than one single quote adjacent to each other. It seems that the replacement will result in only one single quote.

Here is how I tested this:

```
circleci@d6984d82a16b:/$ CONTAINER_DEFS="just test''ing"
circleci@d6984d82a16b:/$ CLEANED_CONTAINER_DEFS=$(echo $CONTAINER_DEFS | sed -E "s/'+/'\\\''/g")
circleci@d6984d82a16b:/$ echo "export CCI_ORB_AWS_ECS_CONTAINER_DEFS='${CLEANED_CONTAINER_DEFS}'" >> /tmp/myscript.sh
circleci@d6984d82a16b:/$ source /tmp/myscript.sh
circleci@d6984d82a16b:/$ echo $CCI_ORB_AWS_ECS_CONTAINER_DEFS
just test'ing
```

Reply:
The suggestion you have provided works best. Let me update.